### DISCUSSION TITLE
Turborepo + AWS lambdas + Serverless

### DISCUSSION THREAD

**Discussion by nezdemkovski** [vercel/turborepo#7009] [Help]
### Describe the feature you'd like to request

A starter example of basic usage of AWS lambdas + serverless framework.

### Describe the solution you'd like

- Turborepo
- Few independent lambdas with serverless files
- Would be nice to use a Prisma in one of the lambdas
- ESLint
- TypeScript

### Describe alternatives you've considered

Currently I use NX on my project and I’m not super happy with it.

**Comment by antoinewg** [comment]
Hello guys, big fan of turborepo. I'm very interested in setting up a basic starter example for backend services, combining those features:
- using the serverless framework
- in a monorepo (to share config/conventions)
- turborepo also provides faster builds and remote caching 💯 
- in typescript
- access to dynamodb or prisma
- consistent formatting and linting (eslint/prettier)
- fast installation (pnpm)
- testing with ~jest~ (vitest)
- using a shared library (for instance for handling errors, or logging)...

I've started here but any help is welcome https://github.com/antoinewg/turbo-serverless-typescript-starter :)

**Comment by stepansib** [comment]
@antoinewg how can we hoist serverless framework and its plugins to the root level? 
Briefly, I dont want to maintain different sls and plugin versions in different services.

**Comment by walterholohan** [comment]
@stepansib did you manage to achieve adding the sls dependecies to the root level?

**Comment by antoinewg** [comment]
@stepansib @walterholohan I've released v1.0.1 of my [template](https://github.com/antoinewg/turboless). Would love some feedback if you have time.

But basically it's now working end to end with caching deployment (and testing included).
Some features coming in the pipeline, but at least it's a working example now. =)

**Comment by anthonyshew** [comment]
(posted prior to converting to Discussion)

Hey, everyone! We've made the decision recently that we're happy with where our examples are at for now. We're not planning on adding to or subtracting from the set we currently have for awhile (if ever).

One thing that we're categorically trying to avoid is that we don't want to have our examples be reaching into the infrastructural side of things. Infrastructure quickly becomes highly use-case-specific, highly variable, and too wide of an area for us to cover from both a knowledge and maintenance standpoint. The examples that we have shortcut you to the point of *applications* rather than *deployments*. 

That said, happy to turn this into a discussion so folks can keep chatting and sharing notes.
